packages/https-proxy/lib/server.coffee

```diff
@@ -69,42 +80,90 @@ class Server
       res.end()
     .pipe(res)
 
+  _upstreamProxyForHostPort: (hostname, port) ->
+    getProxyForUrl("https://#{hostname}:#{port}")
+
   _makeDirectConnection: (req, socket, head) ->
     { port, hostname } = url.parse("http://#{req.url}")
 
+    if upstreamProxy = @_upstreamProxyForHostPort(hostname, port)
+      return @_makeUpstreamProxyConnection(upstreamProxy, socket, head, port, hostname)
+
+    debug("Making direct connection to #{hostname}:#{port}")
     @_makeConnection(socket, head, port, hostname)
 
   _makeConnection: (socket, head, port, hostname) ->
-    cb = ->
+    onConnect = ->
       socket.pipe(conn)
       conn.pipe(socket)
       socket.emit("data", head)
 
       socket.resume()
 
-    ## compact out hostname when undefined
-    args = _.compact([port, hostname, cb])
+    conn = new net.Socket()
+    conn.setNoDelay(true)
 
-    conn = net.connect.apply(net, args)
+    socket.on "close", =>
```

Comment:
Would love to see if we remove this any tests fail.

Are we wired up to the other `close` events? Still not 100% this is tested and the intention of this event is expressed throughout the codebase and/or tests.

Can we explain what situation or why this is necessary or why we're destroying the socket upon receiving this?

Reply:
`socket` is the connection to the destination server, so the intention was to make sure that that connection closes as soon as the client disconnects, because we don't need it anymore.

That being said, I don't think this does anything, since `close` is only emitted after both sides have closed the connection. It can probably be taken out.

I'll add tests to ensure that the socket to the destination server is being closed in the event of the client closing or resetting their connection to the `https-proxy`.

Reply:
Yeah, I added tests that ensure that the the socket to the destination server is closed and they pass with or without this.